src/main/java/org/takes/rs/RsWithCookie.java

```diff
@@ -95,27 +109,33 @@ public RsWithCookie(final Response res, final CharSequence name,
         final CharSequence value, final CharSequence... attrs) {
         super(
             new RsWithHeader(
-                res,
-                "Set-Cookie",
+                new RsWithoutHeader(res, RsWithCookie.SET_COOKIE),
+                RsWithCookie.SET_COOKIE,
                 RsWithCookie.make(
+                    RsWithCookie.getPreviousValue(res),
                     RsWithCookie.checkName(name),
                     RsWithCookie.checkValue(value),
                     attrs
-            )
-        )
+            ))
         );
     }
 
     /**
      * Build cookie string.
+     * @param previous Previous Cookie value.
      * @param name Cookie name
      * @param value Value of it
      * @param attrs Optional attributes, for example "Path=/"
      * @return Text
+     * @checkstyle ParameterNumberCheck (5 lines)
      */
-    private static String make(final CharSequence name,
+    private static String make(final String previous, final CharSequence name,
         final CharSequence value, final CharSequence... attrs) {
-        final StringBuilder text = new StringBuilder(
+        final StringBuilder text = new StringBuilder();
+        if (!previous.isEmpty()) {
+            text.append(String.format("%s,", previous));
+        }
+        text.append(
```

Comment:
@Bertram25 We can pass `String.format` statement below on this line without reaching the line length limit.